src/system/newlib_stubs.c

```diff
@@ -56,3 +59,37 @@ int getentropy(void *_buffer, size_t _length) {
 void __sync_synchronize(void) {
 	__sync_synchronize();
 }
+
+int clock_gettime(clockid_t clock_id, struct timespec* tp) {
+	struct timeval tv;
+	int retval = -1;
+
+	switch (clock_id)
+	{
+	case CLOCK_REALTIME:
+		retval = gettimeofday(&tv, NULL);
+		if (!retval) TIMEVAL_TO_TIMESPEC(&tv, tp);
+		break;
+	
+	default:
+		errno = EINVAL;
+		break;
+	}
+
+	return retval;
+}
+
+extern bool pros::competition::is_connected();
+
+int _gettimeofday(struct timeval* tp, void* tzvp) {
+	if (pros::competition::is_connected()) {
+		tp->tv_sec = vexSystemTimeGet() * 1000;
+		tp->tv_usec = vexSystemHighResTimeGet();
+	}
+	else {
+		tp->tv_sec = _PROS_COMPILE_TIMESTAMP_INT;
+		tp->tv_usec = tp->tv_sec* 1000;
```

Comment:
Why are we passing off a compile time timestamp as a value from a timer? Would it be more appropriate to just error here?

Reply:
```suggestion
		tp->tv_sec = _PROS_COMPILE_TIMESTAMP_INT;
		tp->tv_usec = tp->tv_sec * 1000 * 1000;
```